src/components/LabelSlotsStructure.vue

```diff
@@ -616,6 +628,130 @@ export default Vue.extend({
+                    ? {...(immediatePrevNeighbourSlotInfo as SlotCoreInfos), slotType: (isToPushLeft) ? SlotType.closingBracket : SlotType.openingBracket}
+                    : {...slotInfos, slotId: parentId, slotType: (isToPushLeft) ? SlotType.openingBracket :  SlotType.closingBracket});
+                bracketToPush = document.getElementById(bracketSlotSpanId)?.textContent??"";
+            }
+
+            // 2) if we are after [resp. before] a bracket, now we check if we can push that bracket to the left [resp. right]:
+            // it is possible if the neighbour slot of the bracket is not empty AND not first [resp. last] of its level
+            // OR if we have skipped the first neighbour
+            if(isNextToBracket){
+                if(skipFirstNeighbour){
+                    pushBracket = true;
+                }
+                else {
+                    const neighbourSlot = retrieveSlotFromSlotInfos(neighbourSlotInfosToCheck);
+                    // When we push right, we need to know the neighbour's parent last slot child.
+                    // If the neighbour's slot index isn't showing a bracketed structure, we look directly inside the label slot structure,
+                    // if not, we check with the parent slot (that is then by definition a bracketed structure itself).
+                    const neighbourContainerSlotLastChildIndex  = (neighbourSlotInfosToCheck.slotId.includes(","))
+                        ? (retrieveSlotFromSlotInfos({...neighbourSlotInfosToCheck, slotId: getSlotParentIdAndIndexSplit(neighbourSlotInfosToCheck.slotId).parentId}) as SlotsStructure).fields.length - 1
+                        : this.appStore.frameObjects[neighbourSlotInfosToCheck.frameId].labelSlotsDict[neighbourSlotInfosToCheck.labelSlotsIndex].slotStructures.fields.length - 1; 
+                    const neighbourContainerSlotIndexToCheck = (isToPushLeft) ? 0 : neighbourContainerSlotLastChildIndex;
+                    pushBracket = (neighbourSlot as BaseSlot).code.length > 0 || !new RegExp("^" + neighbourContainerSlotIndexToCheck + "$|," + neighbourContainerSlotIndexToCheck + "$").test(neighbourSlotInfosToCheck.slotId);
+                }
+            }
+            
+            if(pushBracket){
+                // To faciliate the code refactoring, we follow this approach: we clear off the full label slot structure, put the amended content in a new single slot,
+                // and refactor the slots to let our refactoring splitting up operators, brackets etc as normal.
+                let resultingCode = "", newCursorPosition = 0;
+                const neighbourToPushInSlotSpanId = getLabelSlotUID({...slotInfos, slotId: neighbourSlotInfosToCheck.slotId});
+                document.getElementById(this.labelSlotsStructDivId)?.querySelectorAll("." + scssVars.labelSlotInputClassName + ",." + scssVars.labelSlotMediaClassName).forEach((span) => {
```

Comment:
The media slot mention reminds me: does this do the right thing with media literal slots?